yoyodyne/models/__init__.py

```diff
@@ -78,3 +79,16 @@ def add_argparse_args(parser: argparse.ArgumentParser) -> None:
         default="attentive_lstm",
         help="Model architecture. Default: %(default)s.",
     )
+    parser.add_argument(
+        "--tie_embeddings",
+        action="store_true",
+        default=defaults.TIE_EMBEDDINGS,
+        help="Shares embeddings for the source and target vocabularies. "
+        "This should always be set to true for pointer generator models.",
```

Comment:
point of clarification but isn't this just recommended in particular for such models? I think it's valid for the encoder and decoder to have separate embeddings with such models. Putting it another way, when the ptr-gen concept is relevant to your data, so are shared embeddings, but they're not required.

Reply:
Currently it is not technically valid, because our code explicitly combines the source and target dist, and if some tokens are OOV in one or the other (It might only go in one direction, I think the issue is source OOV in target?) we get index errors. We *could* fix this actually by just creating zero indices in the smaller vocabulary I think? But then I guess this breaks things because we need the indices of the output distribution to index the encoded indices.

Maybe we can make a new issue to think this through. This can also be instructive for how to implement an optional mapping for disjoint vocabs.

Reply:
Sorry for the rambling train of though here, but actually I think this PR ensures that "the indices of the output distribution to index the encoded indices" so maybe this is an easy feature update for a different PR.

Reply:
So, we could have separate embeddings with tricks {like} [this]. But I don't think it should give an error if it's not enabled.

If it's mandatory downstream, then we should just force it to be true rather than demand it of our CLI users, I think. 

Sure, create another issue for discussion.